### PR TITLE
[4.0] Fix alerts transparent background

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Testing Instructions
Apply the patch and check the backend when saving an article or saving the global configuration

### Documentation Changes Required
Nope

### CODE REVIEW

File: media/vendor/joomla-custom-elements/js/joomla-alert.js [hunk 1]
```diff
@@ -28,9 +28,9 @@
       this.setAttribute('role', 'alert');
       this.classList.add('joomla-alert--show');
 
-      // If no type has been defined, the default is "info"
-      if (!this.type) {
-        this.setAttribute('type', 'info');
+      // If no level has been defined, the default is "info"
+      if (!this.level || ['info', 'primary', 'warning', 'success', 'danger'].indexOf(this.level) === -1) {
```
Comment (C-Lodder) [inline media/vendor/joomla-custom-elements/js/joomla-alert.js]: I still do not like this `|| ['info', 'primary', 'warning', 'success', 'danger'].indexOf(this.level) === -1`. I dont like the fact we're checking if the user has defined a predefined value.

Should just be:

```
if (!this.level) { ... }
```

**Comment by anuragteapot** [comment]
I have tested this item :white_check_mark: successfully on 1084d206d0c0786b70fe4cc4587f0416e89bec7d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19413">issues.joomla.org/tracker/joomla-cms/19413</a>.</sub>

**Comment by astridx** [comment]
I have tested this item :white_check_mark: successfully on 1084d206d0c0786b70fe4cc4587f0416e89bec7d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19413">issues.joomla.org/tracker/joomla-cms/19413</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19413">issues.joomla.org/tracker/joomla-cms/19413</a>.</sub>

**Comment by ghost** [comment]
Ready to Commit after two successful tests.

**Comment by dgrammatiko** [comment]
I've removed the RTC label as I realised that I was pushing the wrong code here. This PR now restores the alert element to the state of pre Grunt kill PR. As a bonus the alerts on save are green instead of blue.

@Anu1601CS @astridx can you please test again?

**Comment by anuragteapot** [comment]
@dgt41 yeah<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19413">issues.joomla.org/tracker/joomla-cms/19413</a>.</sub>

**Comment by anuragteapot** [comment]
Joomla statistics message still transparent.


screenshot from 2018-01-21 17-00-55


And Popular Articles and Recently Added Articles have no background color.
screenshot from 2018-01-21 17-07-31

**Comment by dgrammatiko** [comment]
@Anu1601CS for popular and Recently there is a bug in the html, we pass `type` attribute but the alerts expect `level`: so this `<joomla-alert type="warning">` will not work but this will `<joomla-alert level="warning">`.
Same for the plugin system/stats

File: media/vendor/joomla-custom-elements/js/joomla-alert.js [hunk 2]
```diff
@@ -1,146 +1,120 @@
 (() => {
   class JoomlaAlertElement extends HTMLElement {
     /* Attributes to monitor */
-    static get observedAttributes() { return ['type', 'dismiss', 'acknowledge', 'href', 'auto-dismiss', 'position', 'textClose', 'textDismiss', 'textAcknowledge']; }
-    get type() { return this.getAttribute('type'); }
-    set type(value) { return this.setAttribute('type', value); }
+    static get observedAttributes() { return ['level', 'dismiss', 'acknowledge', 'href']; }
+    get level() { return this.getAttribute('level') || 'info'; }
```
Comment (dgrammatiko) [inline media/vendor/joomla-custom-elements/js/joomla-alert.js]: well this is stupid as it will always invalidate line 22. The or part here is wrong

**Comment by anuragteapot** [comment]
So , you change <code>level</code> to <code>type</code> .

**Comment by dgrammatiko** [comment]
Yes, it seems that most entries were with attribute type not level so I reverted all the level to type

**Comment by anuragteapot** [comment]
So, all changes are done?. If yes then I can test this patch.

**Comment by dgrammatiko** [comment]
@Anu1601CS go ahead

**Comment by anuragteapot** [comment]
I have tested this item :white_check_mark: successfully on 4c9325efd8c4ac860f5f531c49f95732678b0167<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19413">issues.joomla.org/tracker/joomla-cms/19413</a>.</sub>

**Comment by astridx** [comment]
I have tested this item :white_check_mark: successfully on 4c9325efd8c4ac860f5f531c49f95732678b0167<br /><br />After apply the path: When saving an article or saving the global configuration I see an message with a background that is is not transparent.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19413">issues.joomla.org/tracker/joomla-cms/19413</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19413">issues.joomla.org/tracker/joomla-cms/19413</a>.</sub>

**Comment by C-Lodder** [comment]
@dgt41 am I ever going to be able to pursuade you about that line of code? Lol

**Comment by dgrammatiko** [comment]
not for the current codebase, this is already dead code! @ciar4n and me already working on the new design and the updated specs and I guess we can remove that there. Although alerts *should* strictly be info, warning, success and error, at least in my book...

**Comment by C-Lodder** [comment]
Yes, I agree they should, but what I'm saying is, we should only check if a value doesn't exist exists:

```
if (!newValue) {
```

**not** check if it matches one of our predefined styles:

```
if (!newValue || newValue && ['info', 'warning', 'danger', 'success'].indexOf(newValue) === -1) {
```

File: installation/template/error.php [hunk 3]
```diff
@@ -41,7 +41,7 @@
 				</div>
 				<div id="javascript-warning">
 					<noscript>
-						<joomla-alert level="danger text-center">
```
Comment (wilsonge) [inline installation/template/error.php:44]: type shouldn't contain the class `text-center` right?

Comment (dgrammatiko) [reply installation/template/error.php:44]: true, also that attribute it's not `level` but `type`

**Comment by infograf768** [comment]
@dgt41 
Unsuccessful here for `<joomla-alert type="info">` i.e. the text does not display.
we get a `display:none;`

I do get the text though if I use
`<joomla-alert[type="info"]`
but no colour, no background.

In the image below the blue colour comes from the highlighting

screen shot 2018-01-27 at 10 45 13

**Comment by C-Lodder** [comment]
Looks like the JS isnt being loaded or there's a console log error then. Which view are you testing on?

**Comment by infograf768** [comment]
On a non-multilingual site, I display –as the screenshot shows– the multilangstatus module in its popup.
The js file is loaded
screen shot 2018-01-27 at 11 05 36

Console shows
```
Custom Elements: `Element#attachShadow` was not patched.
webcomponents-ce.min.js:1:7969

```

**Comment by dgrammatiko** [comment]
@infograf768 can you check if the css is loaded? How did you applied the patch?

**Comment by dgrammatiko** [comment]
The warning is ok and it’s coming from the polyfill, so you are testing this with Firefox

**Comment by infograf768** [comment]
Yes I use Firefox as usual

**Comment by C-Lodder** [comment]
If it works with your alternate method then the custom element is defaulting to a span tag, which means the alert-es5.min.js file probably isnt being loaded at all. The console log notice you have comes frim the polyfill

**Comment by dgrammatiko** [comment]
Firefox is not loading es5. The error is either wrong html code (he mentioned that there is a correct tag) or the css. So in this case should be css. Please probably caching (using the old css which expected `level=“info”` instead of `type`

**Comment by infograf768** [comment]
It does not work with the alternate method, it just barely displays the text.

`alert-es5.min.js` (whatever that is) is not loaded indeed
Here is what is loaded:
screen shot 2018-01-27 at 11 32 42


Same result with Safari and Chrome (both could not use a cache as there were used for the first time)

**Comment by dgrammatiko** [comment]
Can you give me steps to replicate this, makes 0 sense trying to guess here

**Comment by infograf768** [comment]
Easy enough
Install a clean site (not multilingual.
In the Module Manager filter by administrator.
Publish the Multilingual Status module

Then click on its icon. It will trigger the popup
screen shot 2018-01-27 at 11 43 08

the code used is
https://github.com/joomla/joomla-cms/blob/4.0-dev/administrator/components/com_languages/tmpl/multilangstatus/default.php#L18-L23

FYI: these are the CSS loaded
screen shot 2018-01-27 at 11 42 54

**Comment by dgrammatiko** [comment]
@infograf768 there you go, apply latest commit:
before
screen shot 2018-01-27 at 11 47 06

after
screen shot 2018-01-27 at 12 00 52


PS the problem was that for the component.php we didn't load the alert element

**Comment by infograf768** [comment]
Works fine now.

**Comment by wilsonge** [comment]
Nice work guys :)

**Pull request merged**